src/iop/grain.c

```diff
@@ -42,6 +42,11 @@
 
 #define GRAIN_SCALE_FACTOR 213.2
 
+#define GRAIN_LUT_SIZE 128
+#define GRAIN_LUT_DELTA_MAX 2.0
+#define GRAIN_LUT_DELTA_MIN 0.005
+#define GRAIN_LUT_PAPER_GAMMA 1.0
+
 #define CLIP(x) ((x < 0) ? 0.0 : (x > 1.0) ? 1.0 : x)
 DT_MODULE_INTROSPECTION(1, dt_iop_grain_params_t)
```

Comment:
Most importantly, you changed the size of params struct, so old history stacks, with params struct of old size are now broken.
Bumping version (1->2) at least prevents them from being broken silently.

As for `DT_MODULE_INTROSPECTION()`, it is just a magic word for introspection generator to create varous functions to be able to generate/modify the params from code easily.